xcell/tests/test_mapper_desy1wl.py

```diff
@@ -29,23 +37,73 @@ def get_es():
                      axis=0).flatten()
 
 
+def test_load_catalog():
+    config = get_config()
+    m = get_mapper()
+    cat = m._load_catalog()
+
+    columns_data = ['coadd_objects_id', 'e1', 'e2',
+                    'psf_e1', 'psf_e2', 'ra', 'dec',
+                    'R11', 'R12', 'R21', 'R22',
+                    'flags_select']
+    columns_zbin = ['zbin_mcal', 'zbin_mcal_1p',
+                    'zbin_mcal_1m', 'zbin_mcal_2p', 'zbin_mcal_2m']
+
+    cat_data = Table.read(config['data_cat'], format='fits')
+    cat_data.keep_columns(columns_data)
+
+    cat_zbin = Table.read(config['zbin_cat'], format='fits')
+    cat_zbin.keep_columns(columns_zbin)
+    cat_data = hstack([cat_data, cat_zbin])
+
+    cat_data.remove_rows(cat_data['dec'] < -90)
+    cat_data.remove_rows(cat_data['dec'] > -35)
+
+    assert(len(cat_data) == len(cat))
+    assert(np.all(cat_data == cat))
+
+
 def test_lite():
     config = get_config()
-    config['path_lite'] = 'xcell/tests/data/'
-    ifile = 0
-    while os.path.isfile(
-            f'xcell/tests/data/DESwlMETACAL_catalog_lite_zbin{ifile}.fits'):
-        os.remove(
-                f'xcell/tests/data/DESwlMETACAL_catalog_lite_zbin{ifile}.fits')
-        ifile += 1
-    xc.mappers.MapperDESY1wl(config)
-    zbin = config['zbin']
-    assert os.path.isfile(
-            f'xcell/tests/data/DESwlMETACAL_catalog_lite_zbin{zbin}.fits')
+    plite = 'xcell/tests/data/'
+    config['path_lite'] = plite
+    remove_lite(plite)
+    # Initialize mapper
+    m = xc.mappers.MapperDESY1wl(config)
+    # Get maps and catalog
+    cat = m.get_catalog()
+    s = m.get_signal_map()
+    psf = m.get_signal_map('PSF')
+    mask = m.get_mask()
 
+    # Check lite files have been created
+    zbin = config['zbin']
+    nside = config['nside']
+
+    for fname in [f'catalog_lite_zbin{zbin}.fits',
+                  f'signal_map_shear_e1_zbin{zbin}_ns{nside}.fits.gz',
+                  f'signal_map_shear_e2_zbin{zbin}_ns{nside}.fits.gz',
+                  f'signal_map_PSF_e1_zbin{zbin}_ns{nside}.fits.gz',
+                  f'signal_map_PSF_e2_zbin{zbin}_ns{nside}.fits.gz',
+                  f'signal_map_PSF_e2_zbin{zbin}_ns{nside}.fits.gz',
+                  f'mask_zbin{zbin}_ns{nside}.fits.gz']:
+        assert os.path.isfile(os.path.join(plite, "DESwlMETACAL_" + fname))
+
+    # Check we recover the same mas and catalog
     # Non-exsisting fits files - read from lite
     config['data_catalogs'] = 'whatever'
-    xc.mappers.MapperDESY1wl(config)
+    m_lite = xc.mappers.MapperDESY1wl(config)
+    cat_from_lite = m_lite.get_catalog()
+    s_from_lite = m_lite.get_signal_map()
+    psf_from_lite = m_lite.get_signal_map('PSF')
+    mask_from_lite = m_lite.get_mask()
+    assert np.all(cat == cat_from_lite)
+    assert np.all(np.array(s) == np.array(s_from_lite))
+    assert np.all(np.array(psf) == np.array(psf_from_lite))
+    assert np.all(mask == mask_from_lite)
```

Comment:
Can you add the coupled nl to this list? That way we check that the `w2s2` map is also read from lite.

Reply:
Good catch. Done.